### PR TITLE
capi: Add a note about error code and returned value relation

### PR DESCRIPTION
**Pull request by gumb0**
Addresses comment https://github.com/wasmx/fizzy/pull/769#discussion_r605143971

### CODE REVIEW

File: include/fizzy/fizzy.h [hunk 1]
```diff
@@ -250,6 +255,9 @@ typedef struct FizzyImportedGlobal
 /// @param  error               Pointer to store detailed error information at. Can be NULL if error
 ///                             information is not required.
 /// @return                     true if module is valid, false otherwise.
+///
+/// @note   FizzyError::code must be ::FizzySuccess if function returns `true` and must not be
```
Comment (gumb0) [inline include/fizzy/fizzy.h:259]: Does this not sound like it's a requirement to a user's input?

Comment (axic) [reply include/fizzy/fizzy.h:259]: Yeah I think so. Would suggest something like "FizzyError::code will be ::FizzySuccess if functions...".

**Review by axic — APPROVED** [review]
Still suggest to use "will be" instead of "must be".

**Review by chfast — APPROVED** [review]

File: include/fizzy/fizzy.h [hunk 2]
```diff
@@ -260,6 +268,9 @@ bool fizzy_validate(
 /// @param  error               Pointer to store detailed error information at. Can be NULL if error
 ///                             information is not required.
 /// @return                     non-NULL pointer to module in case of success, NULL otherwise.
+///
+/// @note   FizzyError::code will be ::FizzySuccess if function returns non-NULL
+///         will and must not be ::FizzySuccess otherwise.
```
Comment (axic) [inline include/fizzy/fizzy.h:273]: Nit: the other places used will not be.

**Pull request merged**